tooling/api/src/http.ts

```diff
@@ -46,515 +46,76 @@
-   *   body: Body.form({
-   *     file: {
-   *       file: '/home/tauri/avatar.png',
-   *       mime: 'image/png',
-   *       fileName: 'avatar.png'
-   *     }
-   *   })
-   * });
-   * ```
-   */
-  async put<T>(
-    url: string,
-    body?: Body,
-    options?: RequestOptions
-  ): Promise<Response<T>> {
-    return this.request({
-      method: 'PUT',
-      url,
-      body,
-      ...options
+      })
     })
   }
 
-  /**
-   * Makes a PATCH request.
-   * @example
-   * ```typescript
-   * import { getClient, Body } from '@tauri-apps/api/http';
-   * const client = await getClient();
-   * const response = await client.patch('http://localhost:3003/users/1', {
-   *   body: Body.json({ email: 'contact@tauri.app' })
-   * });
-   * ```
-   */
-  async patch<T>(url: string, options?: RequestOptions): Promise<Response<T>> {
-    return this.request({
-      method: 'PATCH',
-      url,
-      ...options
-    })
+  interface FetchSendResponse {
+    status: number
+    statusText: string
+    headers: [[string, string]]
+    data: number[]
+    url: string
   }
 
-  /**
-   * Makes a DELETE request.
-   * @example
-   * ```typescript
-   * import { getClient } from '@tauri-apps/api/http';
-   * const client = await getClient();
-   * const response = await client.delete('http://localhost:3003/users/1');
-   * ```
-   */
-  async delete<T>(url: string, options?: RequestOptions): Promise<Response<T>> {
-    return this.request({
-      method: 'DELETE',
-      url,
-      ...options
+  const { status, statusText, url, headers, data } =
+    await invokeTauriCommand<FetchSendResponse>({
+      __tauriModule: 'Http',
+      message: {
+        cmd: 'fetchSend',
+        rid: reqRid
+      }
     })
-  }
-}
 
-/**
- * Creates a new client using the specified options.
- * @example
- * ```typescript
- * import { getClient } from '@tauri-apps/api/http';
- * const client = await getClient();
- * ```
- *
- * @param options Client configuration.
- *
- * @returns A promise resolving to the client instance.
- *
- * @since 1.0.0
- */
-async function getClient(options?: ClientOptions): Promise<Client> {
-  return invokeTauriCommand<number>({
-    __tauriModule: 'Http',
-    message: {
-      cmd: 'createClient',
-      options
-    }
-  }).then((id) => new Client(id))
-}
-
-/** @internal */
-let defaultClient: Client | null = null
-
-/**
- * Perform an HTTP request using the default client.
- * @example
- * ```typescript
- * import { fetch } from '@tauri-apps/api/http';
- * const response = await fetch('http://localhost:3003/users/2', {
- *   method: 'GET',
- *   timeout: 30,
- * });
- * ```
- */
-async function fetch<T>(
-  url: string,
-  options?: FetchOptions
-): Promise<Response<T>> {
-  if (defaultClient === null) {
-    defaultClient = await getClient()
-  }
-  return defaultClient.request({
-    url,
-    method: options?.method ?? 'GET',
-    ...options
+  const res = new Response(Uint8Array.from(data), {
```

Comment:
It seems not suitable for responses with large body size.

Reply:
The current implementation works the same, we currently can only send "normal" arrays to the backend, so unless the new IPC can support streaming bodies, this is what we get. :/